src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -454,30 +454,50 @@ AttributeLocation IAttributeTargetSymbol.AllowedAttributeLocations
             }
         }
 
+#nullable enable
         /// <summary>
         /// Symbol to copy bound attributes from, or null if the attributes are not shared among multiple source parameter symbols.
         /// </summary>
-        /// <remarks>
-        /// Used for parameters of partial implementation. We bind the attributes only on the definition
-        /// part and copy them over to the implementation.
-        /// </remarks>
-        private SourceParameterSymbol BoundAttributesSource
+        private SourceParameterSymbol? BoundAttributesSource
+            => PartialDefinitionPart;
```

Comment:
In every other place we had a `BoundAttributesSource` property, we used the definition part as the source, so I changed it from the implementation to the definition here for consistency.

Also, for attribute ordering, every other position puts the definition part attributes before the impl part attributes. I decided to change it so that parameters also put the definition attributes first. The spec doesn't require any particular ordering (though determinism requires it is always the same for a given program), I just found it easier to make things correct and just a bit cleaner to do it this way.